pkg/observable/channel/map.go

```diff
@@ -80,3 +81,27 @@ func MapReplay[S, D any](
 
 	return dstObservable
 }
+
+// ForEach applies the given forEachFn to each notification received from the
+// observable, similar to map; however, ForEach does not publish to a destination
+// observable. ForEach is useful for side effects.
+func ForEach[V any](
+	ctx context.Context,
+	srcObservable observable.Observable[V],
+	forEachFn ForEachFn[V],
+) {
+	Map(
+		ctx, srcObservable,
+		func(ctx context.Context, src V) (dst V, skip bool) {
+			forEachFn(ctx, src)
+
+			// No downstream observers; MAY always skip.
+			return zeroValue[V](), true
```

Comment:
Shouldn't it republish what `srcObservable` is emitting? This would enable reuse of the resulting observable.